budgetApp1/budgetApp1/SignUpViewController.swift

```diff
@@ -9,46 +9,45 @@ import UIKit
 
 class SignUpViewController: UIViewController {
 
+    // MARK: - Text fields
+    @IBOutlet private weak var fullnameField: UITextField!
+    @IBOutlet private weak var emailAddressField: UITextField!
+    @IBOutlet private weak var usernameField: UITextField!
+    @IBOutlet private weak var passwordField: UITextField!
     
-    @IBOutlet weak var fullnameField: UITextField!
-    @IBOutlet weak var emailAddressField: UITextField!
-    @IBOutlet weak var usernameField: UITextField!
-    @IBOutlet weak var passwordField: UITextField!
+    // MARK: - signUp token
+    private var signedUp = false
     
-    var signedUp = false
-    
-    @IBAction func signedUpClicked(_ sender: Any) {
-        guard let username = usernameField.text else { return }
-        guard let password = passwordField.text else { return }
-        
-        if (username != "" && password != "") {
-            signedUp = true
-            print("user has signed up with ff credentials:", username, password)
-        }
-        else {
-            signedUp = false
-            print("Please sign up")
-        }
+    @IBAction func signedUpClicked(_ sender: Any) { 
+        signedUp = true
     }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
     }
     
+    // MARK: - Segue to validate credentials and log in
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if segue.identifier == "signUpSegue" {
             guard let username = usernameField.text else { return }
             guard let password = passwordField.text else { return }
             guard let fullname = fullnameField.text else { return }
             guard let emailAddress = emailAddressField.text else { return }
             
-            let primaryUser = user(username: username, password: password, fullname: fullname, emailAddress: emailAddress)
+            let primaryUser = User(username: username, password: password, fullname: fullname, emailAddress: emailAddress)
             
-            if let LoginViewController = segue.destination as? LoginViewController {
-                LoginViewController.user1 = primaryUser
-                print("new user:",primaryUser, "has been passed to login")
-                
+            if let loginViewController = segue.destination as? LoginViewController {
+                loginViewController.userTwo = primaryUser
+            } else {
+                let alertController = UIAlertController(
+                                title: "Incomplete user details",
```

Comment:
strings in constant file please. Please keep this in mind so i don't always retype it. All strings should be in. strings or constant file